diff --git a/Contrib/ExDLL/SConscript b/Contrib/ExDLL/SConscript
index 3d2beaf..2652c1a 100755
--- a/Contrib/ExDLL/SConscript
+++ b/Contrib/ExDLL/SConscript
@@ -41,10 +41,10 @@ if env['PLATFORM'] == 'win32':
 else:
 	example += lib_files
 
-	if env.has_key('PREFIX_PLUGINAPI_INC'):
+	if 'PREFIX_PLUGINAPI_INC' in env:
 		env.Distribute(api_files, None, 'pluginapi_inc', '', 'nsis', 'pluginapi', 'pluginapi')
 
-	if env.has_key('PREFIX_PLUGINAPI_LIB'):
+	if 'PREFIX_PLUGINAPI_LIB' in env:
 		env.Distribute(lib, None, 'pluginapi_lib', '', 'nsis', 'pluginapi', 'pluginapi')
 
 env.DistributeExamples(example, path='Plugin')
diff --git a/SCons/Tools/crossmingw.py b/SCons/Tools/crossmingw.py
index a849da0..217a077 100755
--- a/SCons/Tools/crossmingw.py
+++ b/SCons/Tools/crossmingw.py
@@ -85,7 +85,7 @@ def shlib_generator(target, source, env, for_signature):
     implib = env.FindIxes(target, 'LIBPREFIX', 'LIBSUFFIX')
     if implib: cmd.append('-Wl,--out-implib,'+implib.get_string(for_signature))
 
-    def_target = env.FindIxes(target, 'WIN32DEFPREFIX', 'WIN32DEFSUFFIX')
+    def_target = env.FindIxes(target, 'WINDOWSDEFPREFIX', 'WINDOWSDEFSUFFIX')
     if def_target: cmd.append('-Wl,--output-def,'+def_target.get_string(for_signature))
 
     return [cmd]
@@ -109,12 +109,12 @@ def shlib_emitter(target, source, env):
     # or a def file source. There is no option to disable def file
     # target emitting, because I can't figure out why someone would ever
     # want to turn it off.
-    def_source = env.FindIxes(source, 'WIN32DEFPREFIX', 'WIN32DEFSUFFIX')
-    def_target = env.FindIxes(target, 'WIN32DEFPREFIX', 'WIN32DEFSUFFIX')
+    def_source = env.FindIxes(source, 'WINDOWSDEFPREFIX', 'WINDOWSDEFSUFFIX')
+    def_target = env.FindIxes(target, 'WINDOWSDEFPREFIX', 'WINDOWSDEFSUFFIX')
     if not def_source and not def_target:
         target.append(env.ReplaceIxes(dll,  
                                       'SHLIBPREFIX', 'SHLIBSUFFIX',
-                                      'WIN32DEFPREFIX', 'WIN32DEFSUFFIX'))
+                                      'WINDOWSDEFPREFIX', 'WINDOWSDEFSUFFIX'))
     
     return (target, source)
                          
@@ -159,8 +159,8 @@ def generate(env):
     env['AS'] = mingw_prefix + 'as'
     env['AR'] = mingw_prefix + 'ar'
     env['RANLIB'] = mingw_prefix + 'ranlib'
-    env['WIN32DEFPREFIX']        = ''
-    env['WIN32DEFSUFFIX']        = '.def'
+    env['WINDOWSDEFPREFIX']        = ''
+    env['WINDOWSDEFSUFFIX']        = '.def'
     env['SHOBJSUFFIX'] = '.o'
     env['STATIC_AND_SHARED_OBJECTS_ARE_THE_SAME'] = 1
 
