diff --git a/SCons/Config/gnu b/SCons/Config/gnu
index 411f82d..3f330e0 100755
--- a/SCons/Config/gnu
+++ b/SCons/Config/gnu
@@ -82,6 +82,10 @@ stub_env.Append(LINKFLAGS = ['$ALIGN_FLAG'])        # 512 bytes align
 stub_env.Append(LINKFLAGS = ['-Wl,-e,_WinMain@16']) # entry point
 stub_env.Append(LINKFLAGS = ['$MAP_FLAG'])          # generate map file
 
+conf = FlagsConfigure(stub_env)
+conf.CheckCompileFlag('-fno-tree-loop-distribute-patterns')  # don't generate msvcrt!memmove calls
+conf.Finish()
+
 ### makensis environment
 
 makensis_env = defenv.Clone()
@@ -116,6 +120,10 @@ plugin_env.Append(LINKFLAGS = ['-mwindows'])        # build windows executables
 plugin_env.Append(LINKFLAGS = ['$ALIGN_FLAG'])      # 512 bytes align
 plugin_env.Append(LINKFLAGS = ['$MAP_FLAG'])        # generate map file
 
+conf = FlagsConfigure(plugin_env)
+conf.CheckCompileFlag('-fno-tree-loop-distribute-patterns')  # don't generate msvcrt!memmove calls
+conf.Finish()
+
 ### cross-platform util environment
 
 cp_util_env = defenv.Clone()
@@ -144,6 +152,10 @@ if not defenv['DEBUG'] and defenv['STRIP'] and defenv['STRIP_W32']:
 	util_env.Append(LINKFLAGS = ['-s'])                   # strip
 conf.Finish()
 
+conf = FlagsConfigure(util_env)
+conf.CheckCompileFlag('-fno-tree-loop-distribute-patterns')  # don't generate msvcrt!memmove calls
+conf.Finish()
+
 ### cross-platform util environment adjustments
 
 conf = FlagsConfigure(cp_util_env)
