Index: iax-client.h
===================================================================
--- iax-client.h	(revision 8394)
+++ iax-client.h	(working copy)
@@ -94,11 +94,11 @@
 
 #ifdef WIN32
 #if defined(_MSC_VER)
-typedef int (__stdcall *sendto_t)(SOCKET, const char *, int, int, const struct sockaddr *, int);
-typedef int (__stdcall *recvfrom_t)(SOCKET, char *, int, int, struct sockaddr *, int *);
+typedef int (__stdcall *sendto_t)(SOCKET, const char *, int, int, const struct sockaddr *, socklen_t);
+typedef int (__stdcall *recvfrom_t)(SOCKET, char *, int, int, struct sockaddr *, socklen_t *);
 #else
-typedef int PASCAL (*sendto_t)(SOCKET, const char *, int, int, const struct sockaddr *, int);
-typedef int PASCAL (*recvfrom_t)(SOCKET, char *buf, int len, int flags, struct sockaddr *from, int *fromlen);
+typedef int PASCAL (*sendto_t)(SOCKET, const char *, int, int, const struct sockaddr *, socklen_t);
+typedef int PASCAL (*recvfrom_t)(SOCKET, char *buf, int len, int flags, struct sockaddr *from, socklen_t *fromlen);
 #endif
 #else
 typedef int (*sendto_t)(int, const void *, size_t, int, const struct sockaddr *, socklen_t);
Index: iax.c
===================================================================
--- iax.c	(revision 8394)
+++ iax.c	(working copy)
@@ -2984,7 +2984,8 @@
 static struct iax_event *iax_net_read(void)
 {
 	unsigned char buf[65536];
-	int res, sinlen;
+	int res;
+	socklen_t sinlen;
 	struct sockaddr_in sin;
 
 	sinlen = sizeof(sin);
