diff --git a/syncplay/client.py b/syncplay/client.py
index 4e7d61f..ee4dcef 100755
--- a/syncplay/client.py
+++ b/syncplay/client.py
@@ -21,11 +21,10 @@ from twisted.internet import reactor, task, defer, threads
 try:
     SSL_CERT_FILE = None
     import certifi
-    import pem
     from twisted.internet.ssl import Certificate, optionsForClientTLS, trustRootFromCertificates
     certPath = certifi.where()
     if os.path.exists(certPath):
-        SSL_CERT_FILE = certPath
+        os.environ['SSL_CERT_FILE'] = certPath
     elif 'zip' in certPath:
         import tempfile
         import zipfile
@@ -34,7 +33,7 @@ try:
         archive = zipfile.ZipFile(zipPath, 'r')
         tmpDir = tempfile.gettempdir()
         extractedPath = archive.extract(memberPath, tmpDir)
-        SSL_CERT_FILE = extractedPath
+        os.environ['SSL_CERT_FILE'] = extractedPath
 except:
     pass
 
@@ -846,9 +845,10 @@ class SyncplayClient(object):
         port = int(port)
         self._endpoint = HostnameEndpoint(reactor, host, port)
         try:
-            certs = pem.parse_file(SSL_CERT_FILE)
-            trustRoot = trustRootFromCertificates([Certificate.loadPEM(str(cert)) for cert in certs])
-            self.protocolFactory.options = optionsForClientTLS(hostname=host, trustRoot=trustRoot)
+            caCertFP = open(os.environ['SSL_CERT_FILE'])
+            caCertTwisted = Certificate.loadPEM(caCertFP.read().encode('utf-8'))
+            caCertFP.close()
+            self.protocolFactory.options = optionsForClientTLS(hostname=host)
             self._clientSupportsTLS = True
         except Exception as e:
             self.ui.showDebugMessage(str(e))
