diff --git a/Makefile b/Makefile
index bedaac6..e08eb5c 100644
--- a/Makefile
+++ b/Makefile
@@ -30,6 +30,9 @@ endif
 ifeq ($(LOWLEVEL_PACKAGE_MANAGER),pacman-g2)
 	$(INSTALL_DATA) pacman-g2.hook $(DESTDIR)$(etcdir)/pacman-g2/hooks/etckeeper
 endif
+ifeq ($(LOWLEVEL_PACKAGE_MANAGER),ebuild)
+	$(INSTALL_DATA) portage.bashrc $(DESTDIR)$(etcdir)/portage/etckeeper.bashrc
+endif
 ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),yum)
 	$(INSTALL_DATA) yum-etckeeper.py $(DESTDIR)$(prefix)/lib/yum-plugins/etckeeper.py
 	$(INSTALL_DATA) yum-etckeeper.conf $(DESTDIR)$(etcdir)/yum/pluginconf.d/etckeeper.conf
diff --git a/etckeeper.conf b/etckeeper.conf
index f810870..6512800 100644
--- a/etckeeper.conf
+++ b/etckeeper.conf
@@ -1,6 +1,6 @@
 # The VCS to use.
 # VCS="hg"
-VCS="git"
+# VCS="git"
 # VCS="bzr"
 # VCS="darcs"
 
@@ -27,8 +27,8 @@ VCS="git"
 
 # The high-level package manager that's being used.
 # (apt, pacman-g2, yum etc)
-HIGHLEVEL_PACKAGE_MANAGER=apt
+HIGHLEVEL_PACKAGE_MANAGER=emerge
 
 # The low-level package manager that's being used.
 # (dpkg, rpm, pacman-g2, etc)
-LOWLEVEL_PACKAGE_MANAGER=dpkg
+LOWLEVEL_PACKAGE_MANAGER=ebuild
diff --git a/list-installed.d/50list-installed b/list-installed.d/50list-installed
index 177a6c2..cbccf2b 100755
--- a/list-installed.d/50list-installed
+++ b/list-installed.d/50list-installed
@@ -7,4 +7,6 @@ if [ "$LOWLEVEL_PACKAGE_MANAGER" = dpkg ]; then
 		egrep '(ok installed|ok config-files)' | cut -f2,3
 elif [ "$LOWLEVEL_PACKAGE_MANAGER" = rpm ]; then
 	rpm -qa --queryformat "%{name} %{version} %{arch}\n" | sort
+elif [ "$LOWLEVEL_PACKAGE_MANAGER" = ebuild ]; then
+	equery list| sed 's/-\([^-]\+\)\(-r[0-9]\+\)\?$/ \1\2/' | sort
 fi
diff --git a/update-ignore.d/01update-ignore b/update-ignore.d/01update-ignore
index 510e13f..a191eab 100755
--- a/update-ignore.d/01update-ignore
+++ b/update-ignore.d/01update-ignore
@@ -92,6 +92,10 @@ writefile () {
 		ignore "*.pacorig"
 		ignore "*.pacsave"
 		nl
+	elif [ "$LOWLEVEL_PACKAGE_MANAGER" = "ebuild" ]; then
+		comment "new versions of conffiles, stored by ebuild"
+		ignore "._cfg*"
+		nl
 	fi
 	
 	comment "mount(8) records system state here, no need to store these"
