--- adbren.pl.1	2007-06-09 07:29:43.000000000 -0500
+++ adbren.pl	2007-10-15 09:15:45.000000000 -0500
@@ -21,13 +21,11 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
 
-use constant USERNAME => 'username';
-use constant PASSWORD => 'password';
-
 use strict;
 use warnings;
 
 use File::Copy;
+use Getopt::ArgvFile home=>1;
 use Getopt::Long;
 
 my $addmylist = 1;
@@ -37,6 +35,8 @@
 my $format    =
   "\%anime_name_english\%_\%episode\%\%version\%-\%group_short\%.\%filetype\%";
 my $onlyhash = 0;
+my $username;
+my $password;
 
 my $result = GetOptions(
     "addmylist" => \$addmylist,
@@ -45,6 +45,8 @@
     "debug"     => \$debug,
     "format=s"  => \$format,
     "onlyhash"  => \$onlyhash,
+    "username=s"=> \$username,
+    "password=s"=> \$password,
 );
 
 my @files;
@@ -67,8 +69,8 @@
 }
 
 my $a = AniDB::UDPClient->new(
-    username  => USERNAME,
-    password  => PASSWORD,
+    username  => $username,
+    password  => $password,
     client    => "adbren",
     clientver => "3",
     debug     => $debug,
@@ -150,8 +152,7 @@
 	\%anime_type\%, \%anime_name_romaji\%, \%anime_name_kanji\%, 
 	\%anime_name_english\%, \%anime_name_other\%, \%anime_name_short\%, 
 	\%anime_synonyms\%, \%anime_category\%, \%version\%, \%censored\%
-	
-Remember to edit this file to change anidb username and password./;
+/;
     exit;
 }
 
