--- adbren.pl.orig	2008-05-22 21:59:12.000000000 -0500
+++ adbren.pl	2008-05-22 22:03:02.000000000 -0500
@@ -18,17 +18,9 @@
 use strict;
 use warnings;
 use File::Copy;
+use Getopt::ArgvFile home=>1;
 use Getopt::Long;
 
-#################
-# Configuration #
-#################
-use constant USERNAME => 'username';
-use constant PASSWORD => 'password';
-
-# Comment out the line below when you have changed the username password.
-die("You need to change the username password in the file");
-
 # Developers:
 # You can use the AniDB::UDPClient in your own scripts. Just split it out in a seperate file.
 
@@ -42,6 +34,8 @@
   "\%anime_name_english\%_\%episode\%\%version\%-\%group_short\%.\%filetype\%";
 
 my $onlyhash = 0;
+my $username;
+my $password;
 
 my $result = GetOptions(
     "nomylist"  => \$nomylist,
@@ -51,6 +45,8 @@
     "format=s"  => \$format,
     "onlyhash"  => \$onlyhash,
     "nocorrupt" => \$nocorrupt,
+    "username=s"=> \$username,
+    "password=s"=> \$password,
 );
 
 my @files;
@@ -73,8 +69,8 @@
 }
 
 my $a = AniDB::UDPClient->new(
-    username  => USERNAME,
-    password  => PASSWORD,
+    username  => $username,
+    password  => $password,
     client    => "adbren",
     clientver => "4",
     debug     => $debug,
@@ -172,8 +168,6 @@
 	\%anime_type\%, \%anime_name_romaji\%, \%anime_name_kanji\%, 
 	\%anime_name_english\%, \%anime_name_other\%, \%anime_name_short\%, 
 	\%anime_synonyms\%, \%anime_category\%, \%version\%, \%censored\%
-	
-Remember to edit this file to change anidb username and password.
 /;
     exit;
 }
