--- adbren.pl.1	2007-06-09 07:29:43.000000000 -0500
+++ adbren.pl	2007-10-15 16:11:07.000000000 -0500
@@ -1,5 +1,5 @@
 #!/usr/bin/perl -w
-# api.anideb.info batch renamer.
+# api.anideb.net batch renamer.
 
 # Copyright (C) 2006 clip9
 
@@ -238,7 +238,7 @@
     my $class = shift;
     my $self  = bless {}, $class;
 
-    $self->{hostname} = "api.anidb.info";
+    $self->{hostname} = "api.anidb.net";
     $self->{port}     = 9000;
     $self->{handle}   = IO::Socket::INET->new( Proto => 'udp' ) or die($!);
     $self->{ipaddr}   = gethostbyname( $self->{hostname} )
