Index: configure.in
===================================================================
--- configure.in	(revision 11532)
+++ configure.in	(working copy)
@@ -101,12 +101,12 @@
 $PYTHON conftest.py
 has_pygnome=`cat conftest.out`
 rm -f conftest.out conftest.py
+AC_MSG_RESULT($has_pygnome)
 if test NO = "$has_pygnome"
 then
-   AC_MSG_ERROR([
+   AC_MSG_WARN([
 **** The python bindings for gnome2 (gnome2-python) could not be found.])
 fi
-AC_MSG_RESULT($has_pygnome)
 
 AC_MSG_CHECKING(Python bindings for glade)
 cat > conftest.py <<EOF
