diff -ur fontconfig-2.11.1.orig/configure.ac fontconfig-2.11.1/configure.ac
--- fontconfig-2.11.1.orig/configure.ac	2018-04-15 03:01:28.386000000 -0400
+++ fontconfig-2.11.1/configure.ac	2018-04-15 13:10:37.217000000 -0400
@@ -211,6 +211,27 @@
 fi
 AC_CHECK_MEMBERS([struct dirent.d_type],,,
 	[#include <dirent.h>])
+
+# Check the argument type of the gperf hash/lookup function
+AC_MSG_CHECKING([The type of len parameter of gperf hash/lookup function])
+fc_gperf_test="$(echo 'foo' | gperf -L ANSI-C)"
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+	#include <string.h>
+
+	const char *in_word_set(register const char *, register size_t);
+	$fc_gperf_test
+	]])], [FC_GPERF_SIZE_T=size_t],
+	[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+		#include <string.h>
+
+		const char *in_word_set(register const char *, register unsigned int);
+		$fc_gperf_test
+	]])], [FC_GPERF_SIZE_T="unsigned int"],
+	[AC_MSG_ERROR([Unable to determine the type of the len parameter of the gperf hash/lookup function])]
+)])
+AC_DEFINE_UNQUOTED(FC_GPERF_SIZE_T, $FC_GPERF_SIZE_T, [The type of len parameter of the gperf hash/lookup function])
+AC_MSG_RESULT($FC_GPERF_SIZE_T)
+
 #
 # regex
 #
Only in fontconfig-2.11.1: configure.ac.orig
diff -ur fontconfig-2.11.1.orig/src/fcobjs.c fontconfig-2.11.1/src/fcobjs.c
--- fontconfig-2.11.1.orig/src/fcobjs.c	2014-01-20 03:14:20.000000000 -0500
+++ fontconfig-2.11.1/src/fcobjs.c	2018-04-15 13:10:10.867000000 -0400
@@ -25,10 +25,10 @@
 #include "fcint.h"
 
 static unsigned int
-FcObjectTypeHash (register const char *str, register unsigned int len);
+FcObjectTypeHash (register const char *str, register FC_GPERF_SIZE_T len);
 
 static const struct FcObjectTypeInfo *
-FcObjectTypeLookup (register const char *str, register unsigned int len);
+FcObjectTypeLookup (register const char *str, register FC_GPERF_SIZE_T len);
 
 #include "fcobjshash.h"
 
