--- asterisk-1.4.18/configure.orig	2008-01-16 02:13:27.000000000 +0100
+++ asterisk-1.4.18/configure	2008-02-18 10:41:03.000000000 +0100
@@ -19220,8 +19220,13 @@
    if test -f ${IMAP_TK_DIR}/c-client/LDFLAGS ; then
       imap_ldflags=`cat ${IMAP_TK_DIR}/c-client/LDFLAGS`
    fi
-   CPPFLAGS="${CPPFLAGS} -I${IMAP_TK_DIR}/c-client"
-   LIBS="${LIBS} ${IMAP_TK_DIR}/c-client/c-client.a "`echo ${imap_ldflags}`
+#   CPPFLAGS="${CPPFLAGS} -I${IMAP_TK_DIR}/c-client"
+#   LIBS="${LIBS} ${IMAP_TK_DIR}/c-client/c-client.a "`echo ${imap_ldflags}`
+    CPPFLAGS="${CPPFLAGS} -I/usr/include/imap"
+    LIBS="${LIBS} /usr/lib/c-client.a -lssl -lpam"
+    if test -f /usr/bin/krb5-config; then
+     LIBS="${LIBS} "`/usr/bin/krb5-config --libs krb5 gssapi kadm-client kadm-server`
+    fi
    cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
