--- configure	2008-03-11 15:33:45.000000000 +0100
+++ configure.new	2008-03-22 10:43:36.000000000 +0100
@@ -18931,8 +18931,14 @@
 		fi
 		imap_libs="${IMAP_TK_DIR}/c-client/c-client.a"
 	  	imap_include="-I${IMAP_TK_DIR}/c-client"
-      	CPPFLAGS="${CPPFLAGS} ${imap_include}"
-	  	LIBS="${LIBS} ${imap_libs} "`echo ${imap_ldflags}`
+#      	CPPFLAGS="${CPPFLAGS} ${imap_include}"
+#	  	LIBS="${LIBS} ${imap_libs} "`echo ${imap_ldflags}`
+     CPPFLAGS="${CPPFLAGS} -I/usr/include/imap"
+     LIBS="${LIBS} /usr/lib/c-client.a -lssl -lpam"
+     if test -f /usr/bin/krb5-config; then
+      LIBS="${LIBS} "`/usr/bin/krb5-config --libs krb5 gssapi kadm-client kadm-server`
+     fi
+
 	  	cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF

